src/runs.c

```diff
@@ -90,6 +90,41 @@ r_obj* vec_identify_runs(r_obj* x) {
 
 // -----------------------------------------------------------------------------
 
+r_obj* ffi_vec_run_sizes(r_obj* x) {
+  return vec_run_sizes(x);
+}
+
+r_obj* vec_run_sizes(r_obj* x) {
+  const bool start = false;
+  r_obj* where = KEEP(vec_detect_run_bounds0(x, start));
+  const bool* v_where = r_raw_cbegin(where);
+
+  const r_ssize size = r_length(where) / sizeof(bool);
+
+  r_ssize n = 0;
+  for (r_ssize i = 0; i < size; ++i) {
+    n += v_where[i];
+  }
+
+  r_obj* out = KEEP(r_alloc_integer(n));
+  int* v_out = r_int_begin(out);
+  r_ssize j = 0;
+
+  int count = 1;
+
+  for (r_ssize i = 0; i < size; ++i) {
+    const bool end = v_where[i];
```

Comment:
I think `where` should be called `starts`.

And `end` should be called `start` instead? Or am I missing something?

Reply:
https://github.com/r-lib/vctrs/pull/1771#discussion_r1084391910